messaging/activity/FirebaseMessagingActivityGenerator.cs

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System.IO;
+using System.Linq;
+using UnityEngine;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Firebase.Messaging.Editor {
+
+// Handles the generation of the MessagingUnityPlayerActivity java file.
+// Note this regenerates the file every time an Android build occurs,
+// but local changes can be preserved by using the PreserveTag below.
+// This is needed because the source code needs to be present to work across
+// different Unity versions, due to changes in mUnityPlayer.
+// It also adjusts the base class of the file based on if GameActivity is being
+// used (a new feature in Unity 2023).
+public class FirebaseMessagingActivityGenerator : IPreprocessBuildWithReport {
+  private readonly string[] ActivityClassContents = new string[]{
+"/*",
```

Comment:
I tried that originally, the problem I had was consistently locating the template file.  When it is in the Assets folder, that is simple, but it can potentially be in the Packages folder, and I couldn't find a clear way to locate it in both cases.

Reply:
Yeah, I imagine refactoring it will take a bit of time.  I added a TODO, cause I would much rather use a template file in the longer term though.